pkg/gitservice/gitservice.go

```diff
@@ -71,24 +70,23 @@ func (gs *gitService) AddGit(path string) (int, error) {
 	if err != nil {
 		return 0, err
 	}
-	port, err := freeport.GetFreePort()
-	if err != nil {
-		return 0, err
-	}
-
 	r := gitMapping{
-		port:   port,
-		server: &http.Server{Addr: fmt.Sprintf("0.0.0.0:%d", port), Handler: handle},
+		server: &http.Server{Handler: handle},
 		tmpDir: tmpDir,
 	}
 	gs.mappings[path] = &r
+	listener, err := net.Listen("tcp", "0.0.0.0:0")
+	if err != nil {
+		return 0, err
+	}
+	r.port = listener.Addr().(*net.TCPAddr).Port
```

Comment:
I'd suggest checking the type instead of assuming it to avoid potential panics, even though now wouldn't be the case.

Reply:
done, just in case